### PR TITLE
Dev functional API

### PR DESCRIPTION
**Pull request by elephantmipt**
## Before submitting (checklist)

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [ ] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [ ] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [ ] Did you check the docs with `make check-docs`?
- [ ] Did you write any new necessary tests?
- [ ] Did you check that your code passes the unit tests `pytest .` ? 
- [ ] Did you add your new functionality to the docs?
- [ ] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

PS
- [x] I know, that I could [join slack](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for pull request discussion.

### CODE REVIEW

File: catalyst/callbacks/__init__.py [hunk 1]
```diff
@@ -32,5 +32,8 @@
 if IS_PRUNING_AVAILABLE:
```
Comment (Scitator) [inline catalyst/callbacks/__init__.py]: could u please undo changes for this file?

File: catalyst/metrics/functional/segmentation.py [hunk 2]
```diff
@@ -314,7 +314,7 @@ def trevsky(
     """
```
Comment (Scitator) [inline catalyst/metrics/functional/segmentation.py]: could u please undo changes for this file?

File: catalyst/settings.py [hunk 3]
```diff
@@ -37,6 +37,13 @@
 except ModuleNotFoundError:
     IS_XLA_AVAILABLE = False
 
+try:
+    import onnx, onnxruntime  # noqa: F401
```
[the remark this answers is not in the document]
Comment (Scitator) [reply catalyst/settings.py]: add `noqa E401` please

File: catalyst/settings.py [hunk 4]
```diff
@@ -381,6 +388,11 @@ def parse(self) -> dict:
 setattr(SETTINGS, "IS_QUANTIZATION_AVAILABLE", IS_QUANTIZATION_AVAILABLE)  # noqa: B010
 setattr(SETTINGS, "IS_OPTUNA_AVAILABLE", IS_OPTUNA_AVAILABLE)  # noqa: B010
 setattr(SETTINGS, "IS_HYDRA_AVAILABLE", IS_HYDRA_AVAILABLE)  # noqa: B010
+setattr(SETTINGS, "IS_CUDA_AVAILABLE", IS_CUDA_AVAILABLE)  # noqa: B010
+setattr(SETTINGS, "IS_APEX_AVAILABLE", IS_APEX_AVAILABLE)  # noqa: B010
+setattr(SETTINGS, "IS_AMP_AVAILABLE", IS_AMP_AVAILABLE)  # noqa: B010
+setattr(SETTINGS, "NUM_CUDA_DEVICES", NUM_CUDA_DEVICES)  # noqa: B010
+setattr(SETTINGS, "IS_ONNX_AVAILABLE", IS_ONNX_AVAILABLE)
```
[the remark this answers is not in the document]
Comment (Scitator) [reply catalyst/settings.py]: add `# noqa: B010`

File: catalyst/utils/__init__.py [hunk 5]
```diff
@@ -122,4 +122,11 @@
 # )
 
 
+from catalyst.settings import IS_ONNX_AVAILABLE
+
+from catalyst.utils.onnx import convert_to_onnx
```
Comment (Scitator) [inline catalyst/utils/__init__.py]: move under if?

File: catalyst/utils/model_loading.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
+from pathlib import Path
+from typing import Dict
```
[the remark this answers is not in the document]
Comment (Scitator) [reply catalyst/utils/model_loading.py:1]: use `catalyst-make-codestyle --line-length 99` please

File: catalyst/utils/pruning.py [hunk 7]
```diff
@@ -3,16 +3,13 @@
 from torch.nn import Module
 from torch.nn.utils import prune
 
-from catalyst.utils.torch import reset_weights_if_possible
```
Comment (Scitator) [inline catalyst/utils/pruning.py:6]: why have u removed it?

File: catalyst/utils/torch.py [hunk 8]
```diff
@@ -482,6 +482,21 @@ def convert_labels2list(labels: Union[Tensor, List[int]]) -> List[int]:
     return labels_list
 
 
+class ForwardOverrideModel(nn.Module):
```
Comment (Scitator) [inline catalyst/utils/torch.py]: let's move it to `catalyst/tools/{some-name}.py`
and add to the docs

File: catalyst/utils/model_loading.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (elephantmipt) [reply catalyst/utils/model_loading.py:1]: I'm getting this error
`Error: Unsupported flag --line-length`
Even if I use latest version of catalyst-codestyle

**Pull request merged**